src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -197,6 +212,18 @@ var DOMRenderer = ReactFiberReconciler({
     return true;
   },
 
+  commitMount(
+    domElement : Instance,
+    type : string,
+    newProps : Props,
+    rootContainerInstance : Container,
+    internalInstanceHandle : Object,
+  ) : void {
+    if (shouldAutoFocusHostComponent(type, newProps)) {
```

Comment:
This check is still necessary. `commitMount` will be called for host components with refs even if `finalizeInitialChildren` returns `false`. (The "_preserves focus_" test in `ReactDOM-test` covers this case indirectly.)